### DISCUSSION TITLE
Encryption - User-Defined Passwort - can it be changed?

### DISCUSSION THREAD

**Discussion by ghost** [simolus3/drift#1402] [Q&A]
Hello,
I'm writing my first flutter app with moor and encryption using Moor/ffi and sqlcipher_flutter_libs. The phrase for encryption should be a string defined by me + user-defined password. I'm not sure, how I can change the encryption-phrase the user changed the password. Can I first open all tables during function of changing password, let password change and after that encrypt with the new password? I think, maybe the user should close and reopen the app.
Is this alright?
Many thanks for your great job.

**Comment by simolus3** [comment]
Hi,

when using SQLCipher, you can use the `rekey` pragma to change the encryption key. So it's just a matter of `customStatement('PRAGMA rekey = ?', ['your new encryption key'])`. As far as I know, you don't need to close and re-open the database when changing the encryption key.
